library/Booster/Syntax/ParsedKore/Internalise.hs

```diff
@@ -901,11 +901,11 @@ expandAlias alias currentArgs
                     Def.Pattern
                         { term = Util.substituteInTerm substitution term
                         , constraints =
-                            Util.substituteInPredicate substitution <$> constraints
+                            Set.map (Util.substituteInPredicate substitution) constraints
                         }
 
 removeTops :: Def.Pattern -> Def.Pattern
-removeTops p = p{Def.constraints = filter (/= Def.Top) p.constraints}
+removeTops p = p{Def.constraints = Set.filter (/= Def.Top) p.constraints}
```

Comment:
can call `delete` here

Reply:
Good point, there will only be one `Def.Top` in the set now. Will fix it